# Case Study 5- ANOVA

In [1]:
import pandas as pd

df = pd.read_csv('Sales_data_zone_wise.csv')
df

Month  Zone - A  Zone - B  Zone - C  Zone - D
0    Month - 1   1483525   1748451   1523308   2267260
1    Month - 2   1238428   1707421   2212113   1994341
2    Month - 3   1860771   2091194   1282374   1241600
3    Month - 4   1871571   1759617   2290580   2252681
4    Month - 5   1244922   1606010   1818334   1326062
5    Month - 6   1534390   1573128   1751825   2292044
6    Month - 7   1820196   1992031   1786826   1688055
7    Month - 8   1625696   1665534   2161754   2363315
8    Month - 9   1652644   1873402   1755290   1422059
9   Month - 10   1852450   1913059   1754314   1608387
10  Month - 11   2004480   1740365   1317869   1970069
11  Month - 12   1537539   1875658   1237722   1460165
12  Month - 13   1254939   1588473   1348629   1733383
13  Month - 14   1821799   1844081   1617376   1872259
14  Month - 15   1128185   1804613   1767047   1234311
15  Month - 16   1491534   1542053   1882159   1590913
16  Month - 17   1182110   1552388   2234469   2180416
17  Month - 18   1770324   1699213   1249821   2364132
18  Month - 19   1181635   1959268   2009168   1821882
19  Month - 20   1305972   1559670   2107949   2151008
20  Month - 21   1570152   2013615   1540016   2265814
21  Month - 22   1481619   1527574   2255729   2295079
22  Month - 23   1887694   1992155   1348387   1459683
23  Month - 24   1880820   1752873   2098463   2052591
24  Month - 25   1256333   1622671   1521792   1695122
25  Month - 26   1422853   1715465   1853636   1520406
26  Month - 27   1384426   1983163   1611169   1289160
27  Month - 28   1616640   1547991   2128022   2178267
28  Month - 29   1310654   1660092   1947119   1854412

## The mean sales generated by each zone.

In [2]:
df.mean()

Zone - A    1.540493e+06
Zone - B    1.755560e+06
Zone - C    1.772871e+06
Zone - D    1.842927e+06
dtype: float64

Mean sales of Zone A =1540493


Mean sales of Zone B =1755560


Mean sales of Zone C =1772871


Mean sales of Zone D =1842927

## Total sales generated by all the zones for each month

In [3]:
df['TotalSales'] = df.sum(axis=1)
max_value = df['TotalSales']. max()
min_value=df['TotalSales'].min()
max_index=df['TotalSales'].idxmax()
min_index=df['TotalSales'].idxmin()

In [4]:
df[['Month', 'TotalSales']]

Month  TotalSales
0    Month - 1     7022544
1    Month - 2     7152303
2    Month - 3     6475939
3    Month - 4     8174449
4    Month - 5     5995328
5    Month - 6     7151387
6    Month - 7     7287108
7    Month - 8     7816299
8    Month - 9     6703395
9   Month - 10     7128210
10  Month - 11     7032783
11  Month - 12     6111084
12  Month - 13     5925424
13  Month - 14     7155515
14  Month - 15     5934156
15  Month - 16     6506659
16  Month - 17     7149383
17  Month - 18     7083490
18  Month - 19     6971953
19  Month - 20     7124599
20  Month - 21     7389597
21  Month - 22     7560001
22  Month - 23     6687919
23  Month - 24     7784747
24  Month - 25     6095918
25  Month - 26     6512360
26  Month - 27     6267918
27  Month - 28     7470920
28  Month - 29     6772277

In [5]:
print(max_value)
print(min_value)
print(max_index)
print(min_index)

8174449
5925424
3
12


Most sales value is 8174449 in Month 4 and least sales value is 5925424 in Month 13

## Check whether all the zones generate the same amount of sales.

In [6]:
import scipy.stats as stats

stats.f_oneway(df['Zone - A'],df['Zone - B'],df['Zone - C'],df['Zone - D'])

F_onewayResult(statistic=5.672056106843581, pvalue=0.0011827601694503335)

Since pvalue is smaller than alpha value, we reject the null hypothesis

That means amount of sales generated by each zone are different